packages/main/src/resolvers/conversation.ts

```diff
@@ -99,25 +99,20 @@ export const ConversationMutations: ConversationMutationsResolvers = {
 
   async flag(_parent, { ids, isFlagged }) {
     let threads: C.Conversation[] = []
-    ids.forEach(id => {
+    for (const id of ids) {
       const thread = C.mustGetConversation(id)
       updateAction(thread.messages, (accountId, box, uids) => {
         schedule(
-          isFlagged
-            ? actions.unFlag({
-                accountId: String(accountId),
-                box,
-                uids
-              })
-            : actions.flag({
-                accountId: String(accountId),
-                box,
-                uids
-              })
+          actions.setFlagged({
+            accountId: String(accountId),
+            box,
+            uids,
```

Comment:
I should have mentioned earlier - this will flag every message in the conversation. I want to only flag the last message. It is sufficient to take the last value in the `uids` list. A convenient take a list and produce a new list that only keeps the last value is `uids.slice(-1)`.